core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/IsDistinctOverGroupingKeysResult.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.sql.planner.Symbol;
+
+import java.util.List;
+
+public class IsDistinctOverGroupingKeysResult
```

Comment:
Instead of creating a new class for this, let's use `List<Symbol>` or `null` in the recursive static method. Since it's private- we can just document that returning `null` indicates that we are not distinct.

Reply:
Done